server/manager/src/main/java/org/apache/accumulo/manager/Manager.java

```diff
@@ -243,14 +243,16 @@ public synchronized ManagerState getManagerState() {
     return state;
   }
 
-  // ELASTICITIY_TODO it would be nice if this method could take DataLevel as an argument and only
-  // retrieve information about compactions in that data level. Attempted this and a lot of
-  // refactoring was needed to get that small bit of information to this method. Would be best to
-  // address this after issue. May be best to attempt this after #3576.
-  public Map<FateId,Map<String,String>> getCompactionHints() {
+  public Map<FateId,Map<String,String>> getCompactionHints(DataLevel level) {
+    Predicate<TableId> tablePredicate = (tableId) -> true;
+    if (level == DataLevel.ROOT) {
+      tablePredicate = (tableId) -> AccumuloTable.ROOT.tableId().equals(tableId) ? true : false;
+    } else if (level == DataLevel.METADATA) {
+      tablePredicate = (tableId) -> AccumuloTable.METADATA.tableId().equals(tableId) ? true : false;
+    }
```

Comment:
May be able to do this

```suggestion
    Predicate<TableId> tablePredicate = (tableId) -> DataLevel.of(tableId) == level;
```